src/enumerator.rb

```diff
@@ -38,6 +38,20 @@ def to_proc
     end
   end
 
+  def self.produce(*values, &block)
+    if values.size > 1
+      raise ArgumentError, "wrong number of arguments (given #{values.size}, expected 0..1)"
+    end
+
+    new do |yielder|
+      value = values.empty? ? block.call(nil) : values.first
+      while true
```

Comment:
We could use `loop do` here.

Reply:
Maybe I'm just out of the loop (har har) regarding best practices, but those look pretty much like synonyms to me.

Reply:
Someday we can optimize away the truthyness check, then you'd be correct.